internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -449,6 +449,20 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 								),
 							},
 						},
+
+						"default_nginx_controller": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							Default:  managedclusters.NginxIngressControllerTypeAnnotationControlled,
+							ValidateFunc: validation.StringInSlice(
+								[]string{
+									string(managedclusters.NginxIngressControllerTypeAnnotationControlled),
+									string(managedclusters.NginxIngressControllerTypeInternal),
+									string(managedclusters.NginxIngressControllerTypeExternal),
+									string(managedclusters.NginxIngressControllerTypeNone),
```

Comment:
Based on the schema design guidance [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#the-none-value-or-similar), would it make sense to remove the `None` value as an option for this?